File: javascript/fibonacci.js
function fibonacci(num) {
  if (num === 0) return 0
  else if (num === 1) return 1

  let p = [0, 1]
  while (p.length <= num) {
    p.push(p[p.length-1] + p[p.length-2])
  }
  return p[p.length-1]
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(1));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 2");
  console.log("=>", fibonacci(3));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: ruby/fibonacci.rb
# RECURSIVE SOLUTION:

def fibonacci(num)
  return num if num < 2
  return fibonacci(num-1) + fibonacci(num-2)
end

# NON-RECURSIVE SOLUTION:
# def fibonacci(num)
#   return num if num < 2
#   i = 0
#   last_two = [0, 1]
#   until i == num do
#     sum = last_two[0] + last_two[1]
#     last_two = [last_two[1], sum]
#     i += 1
#   end
#   return last_two[0]
# end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution
